portal/views/extend_flask_user.py

```diff
@@ -51,15 +51,19 @@ class LockoutLoginForm(LoginForm):
     def validate(self):
         """prevent locked out users from logging in
 
-        Before verifying the user's credentials
-        check to see if the user is locked out.
-        If the user is locked out display an error
+        If user has exceeded failed attempts, display an error
         message below the password field.
+
         """
+        success = super(LockoutLoginForm, self).validate()
+
         # Find user by email address (email field)
         user_manager = current_app.user_manager
         user = user_manager.find_user_by_email(self.email.data)[0]
 
+        if not success:
+            user.add_password_verification_failure()
```

Comment:
This should only be added when the user_password_failed signal is fired: https://github.com/uwcirg/Flask-User/blob/v0.6.21.1/flask_user/forms.py#L225

Reply:
Never mind, this is better. I was relying too heavily on the signal and should have though more about the desired logic